src/store/proposals/sagas/addProposal.test.ts

```diff
@@ -0,0 +1,557 @@
+  subgraphClientSelector,
+} from '../../app/app.selectors';
+
+import { selectedGovernorSelector } from '../proposals.selectors';
+import { ProposalsActions, proposalsActions } from '../proposals.slice';
+import { ProposalsState } from '../proposals.state';
+
+import { addProposal, checkAddEligibility } from './addProposal';
+import { testAccount } from '../../staking/staking.mock';
+import { userProposalsListQuery } from '../../../queries/proposalListQuery';
+import { convertForMulticall } from '../../utils/utils.sagas';
+
+jest.mock('../../utils/utils.sagas', () => ({
+  ...jest.requireActual('../../utils/utils.sagas'),
+  convertForMulticall: jest.fn(),
+}));
+
+const mockGovernorContract = createMockedContract(
+  GovernorAlpha__factory.connect('0x123', mockSigner),
+  true
+);
+
+const mockStaking = createMockedContract(
+  Staking__factory.connect(constants.AddressZero, mockSigner),
+  false
+);
+
+afterEach(() => {
+  jest.clearAllMocks();
+});
+
+describe('add proposal sagas', () => {
+  const reducer = combineReducers(pick(rootReducer, [Reducers.Proposals]));
+  const initialState: DeepPartial<RootState> = {
+    [Reducers.Proposals]: {
+      ...new ProposalsState(),
+    },
+  };
+
+  describe('addProposal', () => {
+    const initialSteps =
+      initialState[Reducers.Proposals]?.addProposalCall?.steps;
+
+    const mockTx = {
+      hash: '0x01',
+      wait: jest.fn() as ContractTransaction['wait'],
+    } as ContractTransaction;
+
+    const mockTxReceipt = {
+      transactionHash: '0x01',
+    } as TransactionReceipt;
+
+    const successState: DeepPartial<RootState> = {
+      ...initialState,
+      [Reducers.Proposals]: {
+        ...initialState[Reducers.Proposals],
+        addProposalCall: {
+          status: 'success',
+          currentOperation: 'propose',
+          steps: [
+            {
+              ...initialSteps?.[0],
+              tx: mockTx,
+              txReceipt: mockTxReceipt,
+            },
+          ],
+        },
+      },
+    };
+    const errorMsg = 'An unexpected error has occurred. Please try again';
+
+    const failureState: DeepPartial<RootState> = {
+      ...initialState,
+      [Reducers.Proposals]: {
+        ...initialState[Reducers.Proposals],
+        addProposalCall: {
+          status: 'failure',
+          steps: [
+            {
+              ...initialSteps?.[0],
+              error: errorMsg,
+            },
+          ],
+          currentOperation: 'propose',
+        },
+      },
+    };
+
+    const failureWalletNotConnectedState: DeepPartial<RootState> = {
+      ...initialState,
+      [Reducers.Proposals]: {
+        ...initialState[Reducers.Proposals],
+        addProposalCall: {
+          status: 'failure',
+          steps: [
+            {
+              ...initialSteps?.[0],
+              error: 'Wallet not connected',
+            },
+          ],
+        },
+      },
+    };
+
+    const proposalGovernorAdminPayload: ProposalsActions['addProposal'] = {
```

Comment:
Maybe like that?
```suggestion
    const mockAddProposalValues: AddProposalFields = {
        SendProposalContract: GovernorTypes.GovernorAdmin,
        Description: 'test description of proposal',
     ...
```

And then:
```typescript
const getBasePath = () =>
      expectSaga(addProposal, proposalActions.addProposal(mockAddProposalValues))
```